src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,846 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
         {
             _output = output;
-            _testFixture = testFixture;
+        }
+
+        private void Run(Action<DefaultAnalyzerAssemblyLoader, AssemblyLoadTestFixture> action, [CallerMemberName] string? memberName = null)
+        {
+#if NETCOREAPP
+            var alc = AssemblyLoadContextUtils.Create($"Test {memberName}");
+            var assembly = alc.LoadFromAssemblyName(typeof(AssemblyLoadContextInvokeUtil).Assembly.GetName());
+            var util = assembly.CreateInstance(typeof(AssemblyLoadContextInvokeUtil).FullName)!;
+            var method = util.GetType().GetMethod("Exec", BindingFlags.Public | BindingFlags.Instance)!;
+            method.Invoke(util, new object[] { action.Method.DeclaringType!.FullName!, action.Method.Name });
+
+#else
+            AppDomain? appDomain = null;
+            try
+            {
+                appDomain = AppDomainUtils.Create($"Test {memberName}");
+                var type = typeof(AppDomainInvokeUtil);
+                var util = (AppDomainInvokeUtil)appDomain.CreateInstanceAndUnwrap(type.Assembly.FullName, type.FullName);
+                util.Exec(action.Method.DeclaringType.FullName, action.Method.Name);
+            }
+            finally
+            {
+                AppDomain.Unload(appDomain);
+            }
+#endif
+        }
+
+        /// <summary>
+        /// This is called from our newly created AppDomain or AssemblyLoadContext and needs to get 
+        /// us back to the actual test code to execute. The intent is to invoke the lambda / static
+        /// local func where the code exists.
+        /// </summary>
+        private static void InvokeTestCode(string typeName, string methodName)
+        {
+            var type = typeof(DefaultAnalyzerAssemblyLoaderTests).Assembly.GetType(typeName, throwOnError: false)!;
+            var member = type.GetMethod(methodName, BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance)!;
+
+            // A static lambda will still be an instance method so we need to create the closure
+            // here.
+            var obj = member.IsStatic
+                ? null
+                : type.Assembly.CreateInstance(typeName);
+
+            using var fixture = new AssemblyLoadTestFixture();
+            var loader = new DefaultAnalyzerAssemblyLoader();
+            member.Invoke(obj, new object[] { loader, fixture });
         }
 
         [Fact, WorkItem(32226, "https://github.com/dotnet/roslyn/issues/32226")]
         public void LoadWithDependency()
         {
-            var analyzerDependencyFile = _testFixture.AnalyzerDependency;
-            var analyzerMainFile = _testFixture.AnalyzerWithDependency;
-            var loader = new DefaultAnalyzerAssemblyLoader();
-            loader.AddDependencyLocation(analyzerDependencyFile.Path);
+            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
```

Comment:
Did you verify that when the delegate throws an exception that it propagates correctly back to here?

Reply:
Indeed yes I verified the exceptions propagate, trigger errors and have meaningful error messages in the test explorer. That's how I went about fixing the broken tests that I found.

Reply:
Ah I see the confusion. I after writing the code to do this, and using it to debug test failures, I accidentally deleted it when cleaning up before pushing. Will add back shortly.